src/analysis/imports.rs

```diff
@@ -126,13 +125,36 @@ impl Imports {
     fn common_checks(&self, name: &str) -> bool {
         if name == "crate::ffi" || (!name.contains("::") && name != "xlib") {
             false
-        } else if let Some(ref defined) = self.defined {
-            !((name.starts_with("crate::") && &name[7..] == defined) || name == defined)
+        } else if self.defined.contains(name) {
+            false
+        } else if let Some(name) = name.strip_prefix("crate::") {
+            !self.defined.contains(name)
         } else {
             true
         }
     }
 
+    /// Declares that `name` is defined in scope
+    ///
+    /// Removes existing imports from `self.map` and marks `name` as
+    /// available to counter future import "requests".
+    pub fn add_defined(&mut self, name: &str) {
+        // Always store the name without crate:: prefix
+        let name = if let Some(name_without_crate) = name.strip_prefix("crate::") {
+            // Remove crate::<name>:
+            self.map.remove(name);
+            name_without_crate
+        } else {
+            // Take extra care to remove an eventual use crate::<name>; as well:
+            self.map.remove(&format!("crate::{}", name));
+            name
+        };
```

Comment:
Welp edition 2018 `crate::` handling takes quite a bit more effort than expected. I don't want to make the assumption that one or the other format stored in here, but can we perhaps make it as such?

Reply:
The problem is that currently names are stored in both variants? That's annoying and should ideally be cleaned up to only use the 2018 variant...

That should also clean up such handling elsewhere.

Reply:
I have to investigate this more carefully but it is quite possible that both variants are used intertwined, depending on where they are called from.

After all, needing an external symbol from outside the file is going to be imported with `use crate::MySymbol;`, but when defined it's going to be defined and used as `MySymbol` (though, supposedly, also reachable through `crate::MySymbol` even though that's a confusing).

I guess it is much easier for me to solve this by simply calling `add_defined` with a `crate::` prefixed variant after which everything should be working fine.